### PR TITLE
Upgrade deps : sqlalchemy 2.0.0 major migration and others

### PR DESCRIPTION
**Pull request by yashvardhannanavati**

### CODE REVIEW

File: iib/web/models.py [hunk 1]
```diff
@@ -212,10 +210,10 @@ class RequestArchitecture(db.Model):
 
     # A primary key is required by SQLAlchemy when using declaritive style tables, so a composite
     # primary key is used on the two required columns
-    request_id = db.Column(
+    request_id: Mapped[int] = db.mapped_column(
         db.Integer, db.ForeignKey('request.id'), autoincrement=False, index=True, primary_key=True
```
Comment (xDaile) [inline iib/web/models.py:214]: Is this db.Integer parameter needed with Mapped[int]?

File: iib/web/models.py [hunk 2]
```diff
@@ -355,34 +353,34 @@ class Request(db.Model):
...
-        db.Integer, db.ForeignKey('request_state.id'), index=True, unique=True
+    batch_id: Mapped[int] = db.mapped_column(db.ForeignKey('batch.id'), index=True)
+    batch: Mapped['Batch'] = db.relationship('Batch', back_populates='requests')
+    request_state_id: Mapped[int] = db.mapped_column(
+        db.ForeignKey('request_state.id'), index=True, unique=True
     )
     # This maps to a value in RequestTypeMapping
-    type = db.Column(db.Integer, nullable=False)
-    user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
+    type: Mapped[int]
```
Comment (xDaile) [inline iib/web/models.py:366]: Missing db.mapped_column(), or it is not needed here?

**Review by xDaile — DISMISSED** [review]

File: iib/web/models.py (hunk 1 above)
Comment (yashvardhannanavati) [reply iib/web/models.py]: thanks for catching this. I have updated such instances.

**Comment by yashvardhannanavati** [comment]
@lipoja @xDaile I have tested the change locally and was successfully able to execute add/rm/merge-index-image requests. @lipoja could you please help test fbc-operations endpoint? For some reason, quay is rate-limiting me.

**Comment by lipoja** [comment]
@yashvardhannanavati I've tested fbc-operations locally and it works

**Review by lipoja — APPROVED** [review]

**Review by xDaile — APPROVED** [review]

**Pull request merged**